### PR TITLE
examples/evm-multichain-dapp

### PR DESCRIPTION
**Pull request by saurabhburade**
## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm lint`
- [x] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

The template repository can be found [here](https://github.com/saurabhburade/with-evm-chains)
Public reviews, and improvement suggestions are welcome!
## Supported Blockchains
1. ETHEREUM
2. ROPSTEN
3. RINKEBY
4. GÖRLI
5. KOVAN
6. MATIC
7. MATIC_TESTNET
8. FANTOM
9. FANTOM_TESTNET
10. XDAI
11. BSC
12. BSC_TESTNET
13. ARBITRUM
14. ARBITRUM_TESTNET
15. MOONBEAM_TESTNET
16. AVALANCHE
and more

### Note 
- This template has the flexibility to seamlessly extend to other EVM Blockchains.
- It has web3.js and wallet connection modules.
- It supports blockchain Read and Write functionalities.

### CODE REVIEW

File: examples/evm-multichain-dapp/next-env.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/// <reference types="next" />
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/next-env.d.ts:1]: This file should be in `.gitignore`, please remove.

File: examples/evm-multichain-dapp/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "evm-template",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "start": "next start",
+    "lint": "next lint"
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/package.json:9]:
```suggestion
    "start": "next start"
```

File: examples/evm-multichain-dapp/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "web3": "^1.7.4"
+  },
+  "devDependencies": {
+    "@types/node": "18.6.2",
+    "@types/react": "18.0.15",
+    "@types/react-dom": "18.0.6",
+    "eslint": "8.20.0",
+    "eslint-config-next": "12.2.3",
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/package.json:27]:
```suggestion
```

File: examples/evm-multichain-dapp/package.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@web3-react/abstract-connector": "^6.0.7",
+    "@web3-react/core": "^6.1.9",
+    "@web3-react/injected-connector": "^6.0.7",
+    "@web3-react/walletconnect-connector": "^6.2.13",
+    "bignumber.js": "^9.0.2",
+    "next": "12.2.3",
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/package.json:17]:
```suggestion
    "next": "latest",
```

File: examples/evm-multichain-dapp/package.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "evm-template",
+  "version": "0.1.0",
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/package.json:3]:
```suggestion
```

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, please revert the `pnpm-lock.yaml` changes, as examples should not modify the lock file.

Can you also address my comments? Make sure you read through https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md to not miss any necessary changes!

File: examples/evm-multichain-dapp/package.json (hunk 2 above)
Comment (saurabhburade) [reply examples/evm-multichain-dapp/package.json]: - Made suggested changes.
- Merged [vercel:canary](https://github.com/vercel/next.js) to [saurabhburade:example/evm-dapp](https://github.com/saurabhburade/next.js/tree/example/evm-dapp) for latest implementation of pnpm-lock.yaml file.

File: examples/evm-multichain-dapp/next-env.d.ts (hunk 1 above)
Comment (saurabhburade) [reply examples/evm-multichain-dapp/next-env.d.ts:1]: Added ` .gitignore` file

File: pnpm-lock.yaml [hunk 6]
```diff
@@ -30872,4 +30872,4 @@ packages:
       }
     name: ci-info
     version: 2.0.0
-    dev: true
+    dev: true
```
Comment (balazsorban44) [inline pnpm-lock.yaml:30875]:
```suggestion
    dev: true

```

File: examples/evm-multichain-dapp/config/blockchains.json [hunk 7]
```diff
@@ -0,0 +1,8778 @@
+{
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/config/blockchains.json:1]: This file is huge, I wonder if we can do anything about it? Is this generated/downloaded from somewhere? Maybe we could add instructions on how to download it. What do you think?

File: examples/evm-multichain-dapp/pages/index.tsx [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+const Home: NextPage = () => {
+  const { account, activate, chainId, deactivate } = useWeb3React()
+  const [nativeBalance, setNativeBalance] = useState(0)
+  const [sendAddress, setSendAddress] = useState('')
+  const [sendAmount, setSendAmount] = useState('0')
+
+  const { web3 } = useWeb3()
+  // @ts-ignore TYPE NEEDS FIXING
+  const chainData = blockchains[chainId]
```
Comment (balazsorban44) [inline examples/evm-multichain-dapp/pages/index.tsx:22]: Looks like we are pulling in a JSON file with ~9000 lines here. Is this really optimal? :thinking:

Maybe we could use `getStaticProps` and a [dynamic route](https://nextjs.org/docs/routing/dynamic-routes) with `fallback: false`.

Eg.:

`/blockchains/1`

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
I had one concern. Can we discuss before merging?

File: examples/evm-multichain-dapp/config/blockchains.json (hunk 7 above)
Comment (saurabhburade) [reply examples/evm-multichain-dapp/config/blockchains.json:1]: It is derived from https://chainid.network/chains.json 
Paired it with the key as `chainId`
Like this `chainId:{...BLOCKCHAIN DATA ...}`

File: examples/evm-multichain-dapp/pages/index.tsx (hunk 8 above)
Comment (saurabhburade) [reply examples/evm-multichain-dapp/pages/index.tsx:22]: - We can reduce the file size to ensure data is available for only listed blockchains `config/chainIds.ts`.
- Agree, the dynamic route & getStaticProps can be implemented to optimize it according to chainId.

**Review by balazsorban44 — APPROVED** [review]
Thanks!

**Comment by saurabhburade** [comment]
Thank You [balazsorban44](https://github.com/balazsorban44) for your time & suggestions 🍻.

**Pull request merged**